spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/sampler/UuidToBigIntegerConverterTest.java

```diff
@@ -0,0 +1,52 @@
+package org.springframework.cloud.sleuth.sampler;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import lombok.Data;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.math.BigInteger;
+import java.util.UUID;
+
+import static java.math.BigInteger.ONE;
+import static java.math.BigInteger.ZERO;
+import static junitparams.JUnitParamsRunner.$;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.springframework.cloud.sleuth.sampler.UuidToBigIntegerConverterTest.TestData.sample;
+
+@RunWith(JUnitParamsRunner.class)
+public class UuidToBigIntegerConverterTest {
+
+	private static final BigInteger TWO = new BigInteger("2");
+
+	@Test
+	@Parameters
+	public void should_convert_uuid_to_big_integer(TestData testData) throws Exception {
+		UUID uuid = new UUID(testData.mostSignificantBits, testData.leastSignificantBits);
+
+		BigInteger bigInteger = UuidToBigIntegerConverter.uuidToBigInt(uuid);
+
+		then(bigInteger).isEqualByComparingTo(testData.expectedInteger);
+	}
+
+	public Object[] parametersForShould_convert_uuid_to_big_integer() {
+		return $(
+				sample(Long.MAX_VALUE, Long.MAX_VALUE, TWO.pow(126).subtract(ONE)),
```

Comment:
style + tiredness, so feel free to say "I'll pass"..

because I'm tired I can't remember why 2^126-1... to relate the left side to the right side it might be clearer to use a byte buffer.  ex. this is longer, but doesn't require the reader to think through the math..

```java
new BigInteger(
  ByteBuffer.allocate(128).putLong(Long.MAX_VALUE).putLong(Long.MAX_VALUE).array()
)
```
Above syntax might be wrong.. even easier with Okio buffer as you don't need to pre-allocate